akka-http-tests/src/test/scala/akka/http/scaladsl/coding/DeflateSpec.scala

```diff
@@ -24,5 +33,20 @@ class DeflateSpec extends CoderSpec {
         ourDecode(ByteString(0, 1, 2, 3, 4))
       }).ultimateCause should be(a[DataFormatException])
     }
+    "properly round-trip encode/decode an HttpRequest using wrapping" in {
+      val request = HttpRequest(POST, entity = HttpEntity(largeText))
```

Comment:
Would be nice to add a few tests for the edge cases like the no compression deflate compression without wrapping case discussed in the PR.